src/Language/PureScript/CodeGen/JS.hs

```diff
@@ -229,20 +229,16 @@ moduleToJs (Module _ coms mn _ imps exps reExps foreigns decls) foreignInclude =
        else AST.Comment (AST.SourceComments com) <$> nonRecToJS a i (modifyAnn removeComments e)
   nonRecToJS (ss, _, _, _) ident val = do
     js <- valueToJs val
-    withPos ss $ AST.VariableIntroduction Nothing (identToJs ident) (Just js)
-
-  withPos :: SourceSpan -> AST -> m AST
-  withPos ss js = do
-    withSM <- asks (elem JSSourceMap . optionsCodegenTargets)
-    return $ if withSM
-      then withSourceSpan ss js
-      else js
+    return $ AST.VariableIntroduction (Just ss) (identToJs ident) (Just js)
```

Comment:
I have two comments on this (mainly for myself). 
First, `withPos` was originally called recursively by `valueToJs`. In other words, we were checking this CLI flag every time, rather than checking it once and determining whether to use a path that doesn't add annotations versus one that does.

Second, by removing this completely, we always generate source spans, even when it's not requested by the user as a codegen target. While the AST may have source spans, we still don't generate the source maps unless it's listed as a codegen target.

Reply:
This is something I was debating if I should change or not. 
Most of the code in `JS.hs` where `withPos` makes sense to be used already ignored it and simply passed `Just ss` instead. 
So I decided for cleaning up a little bit and removing it completely (IIRC there was only one place where it was still in use after my relevant changes). And all these SourceSpans are completely ignored in the end if you don't compile with sourcemaps (IIRC you can verify that looking for the `Emit` instances in `Printer.hs`). 

The tradeoffs here would be, as far as I can see, a (negligible) extra memory use by passing along `Just ss` instead of `Nothing`, for a (negligible) speed-up in compilation time for not having to run `withPos` every time.